Realm/RLMAnalytics.mm

```diff
@@ -194,73 +296,206 @@ static auto RLMSysCtl(int *mib, u_int mibSize, size_t *bufferSize) {
+        // AppCode is using.
+        @"IDE Version": RLMXCodeVersion() ?: kUnknownString,
+    };
 }
 
-void RLMSendAnalytics() {
+// This will only be executed once but depending on the number of objects, could take sometime
+static NSDictionary *RLMSchemaMetrics(RLMSchema *schema) {
+    NSDictionary *dict = @{
+        @"Embedded_Object": @0,
+        @"Asymmetric_Object": @0,
+        @"Reference_Link": @0,
+        @"Mixed": @0,
+        @"Primitive_List": @0,
+        @"Primitive_Set": @0,
+        @"Primitive_Dictionary": @0,
+        @"Reference_List": @0,
+        @"Reference_Set": @0,
+        @"Reference_Dictionary": @0,
+    };
+    NSMutableDictionary *featuresDictionary = [[NSMutableDictionary alloc] init];
+    [featuresDictionary addEntriesFromDictionary: dict];
+    for (RLMObjectSchema *objectSchema in schema.objectSchema) {
+        if (objectSchema.isEmbedded) {
+            [featuresDictionary setObject:@1 forKey:@"Embedded_Object"];
+        }
+        if (objectSchema.isAsymmetric) {
+            [featuresDictionary setObject:@1 forKey:@"Asymmetric_Object"];
+        }
+
+        for (RLMProperty *property in objectSchema.properties) {
+            if (property.array) {
+                if (property.type == RLMPropertyTypeObject) {
+                    [featuresDictionary setObject:@1 forKey:@"Reference_List"];
+                } else {
+                    [featuresDictionary setObject:@1 forKey:@"Primitive_List"];
+                }
+                continue;
+            }
+            if (property.set) {
+                if (property.type == RLMPropertyTypeObject) {
+                    [featuresDictionary setObject:@1 forKey:@"Reference_Set"];
+                } else {
+                    [featuresDictionary setObject:@1 forKey:@"Primitive_Set"];
+                }
+                continue;
+            }
+            if (property.dictionary) {
+                if (property.type == RLMPropertyTypeObject) {
+                    [featuresDictionary setObject:@1 forKey:@"Reference_Dictionary"];
+                } else {
+                    [featuresDictionary setObject:@1 forKey:@"Primitive_Dictionary"];
+                }
+                continue;
+            }
+
+            switch (property.type) {
+               case RLMPropertyTypeAny:
+                    [featuresDictionary setObject:@1 forKey:@"Mixed"];
+                  break;
+               case RLMPropertyTypeObject:
+                    [featuresDictionary setObject:@1 forKey:@"Reference_Link"];
+                  break;
+                case RLMPropertyTypeLinkingObjects:
+                     [featuresDictionary setObject:@1 forKey:@"Backlink"];
+                   break;
+               default:
+                    break;
+            }
+        }
+    }
+    return featuresDictionary;
+}
+
+static NSDictionary *RLMConfigurationMetrics(RLMRealmConfiguration *configuration) {
+    BOOL isSyncEnable = configuration.syncConfiguration != nil;
```

Comment:
Every read of the syncConfiguration property is allocating a new object, which should be cached.

BOOL is a bad obj-c type (it's a typedef for `char` rather than `bool`) and should generally only appear at public API boundaries.